controllers/k8ssandra/k8ssandracluster_controller.go

```diff
@@ -170,8 +170,8 @@ func (r *K8ssandraClusterReconciler) SetupWithManager(mgr ctrl.Manager, clusters
 	clusterLabelFilter := func(mapObj client.Object) []reconcile.Request {
 		requests := make([]reconcile.Request, 0)
 
-		kcName := utils.GetLabel(mapObj, api.K8ssandraClusterNameLabel)
-		kcNamespace := utils.GetLabel(mapObj, api.K8ssandraClusterNamespaceLabel)
+		kcName := utils.GetLabel(mapObj, utils.K8ssandraClusterNameLabel)
```

Comment:
I realize that this is subjective, but I am not fan of moving the labels and annotations constants into the `utils` package. It seems fairly common for them to be defined with the api types. cass-operator is of course an example of this. https://github.com/kubernetes-sigs/cluster-api is a really good example because it is a large project with lots of type, and they are documented really well in the source code. The api types provide documentation as do label and annotation constant. It seem less obvious to me (again my opinion) to look in a utils package common, well-known labels and annotations. If we want to move them out of the apis directory, I would prefer a dedicated package, maybe `constants`.

Reply:
The problem is that if `utils` references `api`, then `api` cannot reference `utils`, not even transitively.

Currently we have: `api` -> `images` -> `utils`.

Looking at cluster-api, I see that they avoid the circular reference by placing their labels and annotation utilities in separate packages. I will give this idea a try.